hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java

```diff
@@ -476,6 +476,9 @@ public Builder setDatanodeDetails(DatanodeDetails details) {
       this.setupTime = details.getSetupTime();
       this.revision = details.getRevision();
       this.buildDate = details.getBuildDate();
+      this.persistedOpState = details.getPersistedOpState();
+      this.persistedOpStateExpiryEpochSec =
+          details.getPersistedOpStateExpiryEpochSec();
       return this;
```

Comment:
These fields had been missed from the DatanodeDetails builder object, so when the DN reported back its "persisted state" the DN was always IN_SERVICE. Adding this change fixed that problem and allowed the state to be returned correctly.